src/straPy/data_vis.py

```diff
@@ -1,29 +1,62 @@
-def density_ci_plot(dist, ci):
-    """Makes a density plot with its confidence interval
-
-    The plot will show a density curve of a sampling distribution 
-    `dist` with its confidence interval `ci`.
-
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+import warnings
 
+def histogram_ci_plot(sample, rep, bin_size = 30, n="auto", ci_level=0.95, ci_random_seed=None, title = "", x_axis = "Bootstrapped Sample Mean", y_axis = "Count"):
+    
+    """Makes a histogram of a bootstrapped sampling distribution 
+    with its confidence interval and observed mean.
+     
     Parameters
     ----------
-    dist : numpy.ndarray
-        bootstrapped sampling distribution
-    ci : numpy.ndarray
-        confidence interval
+    sample : list or numpy.ndarray or pandas.core.series.Series 
+        sample to bootstrap
+    rep : int
+        number of replicates of the distribution
+    bin_size = int
+        a number of bins representing intervals of equal size over the range
+    n : str or int, default="auto"
+        bootstrap sample size, "auto" specifies using the same size as the sample
+    ci_level : float, default=0.95
+        confidence level
+    ci_random_seed : None or int, default=None
+        seed for random state
+    title : str, default = ""
+        title of the histogram
+    x_axis : str, default = "Bootstrap Sample Mean"
+        name of the x axis
+    y_axis : str, default = "Count"
+        name of the y axis
     
-
     Returns
     -------
-    plot object
-        density plot with confidence interval
+    plot: histogram
+        histogram of bootstrapped distribution with confidence interval and observed mean
     
     Examples
     --------
-    >>> density_ci_plot([1, 1, 2, 3, 5, 10]，[1., 9.375]）
+    >>> histogram_ci_plot([1, 2, 3, 4, 5, 6, 7], 1000 ,n=100, ci_level=0.95, ci_random_seed=123)
     """
-    pass
 
+    if not isinstance(title, str):
+        raise TypeError("The value of the argument 'title' must be type of str.")
+        
+    if not isinstance(x_axis, str):
+        raise TypeError("The value of the argument 'x_axis' must be type of str.")
+        
+    if not isinstance(y_axis, str):
+        raise TypeError("The value of the argument 'y_axis' must be type of str.")
+        
+    plt.hist(calculate_boot_stats(sample, rep, level=ci_level, random_seed = ci_random_seed, pass_dist=True)[1], density=False, bins=bin_size)
+    plt.axvline(calculate_boot_stats(sample, rep, level=ci_level, random_seed = ci_random_seed, pass_dist=True)[0]["lower"], color='k', linestyle='--')
+    plt.axvline(calculate_boot_stats(sample, rep, level=ci_level, random_seed = ci_random_seed, pass_dist=True)[0]["sample_mean"], color='r', linestyle='-')
+    plt.axvline(calculate_boot_stats(sample, rep, level=ci_level, random_seed = ci_random_seed, pass_dist=True)[0]["upper"], color='k', linestyle='--')
```

Comment:
@zackt113, just a small formatting change, the code lines according to pep8 standard should not be more than 79 characters.